src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3982,7 +3982,7 @@ public String descriptorString() {
      */
     @Override
     public Class<?> componentType() {
-        return isArray() ? componentType : null;
+        return componentType;
```

Comment:
Did you consider calling getComponentType(), which already returns the field?

Are there regression tests for Class::componentType()/Class::getComponentType() somewhere? I didn't see them in the obvious test/jdk/java/lang/Class directory.

Reply:
> Did you consider calling getComponentType(), which already returns the field?

I don't think we want an extra indirection here - The logic here is quite simple. Counterargument could be that `MethodType::descriptorString` calls `toMethodDescriptorString`, except these methods involve more complex caching and is not simple like this field access.

> Are there regression tests for Class::componentType()/Class::getComponentType() somewhere? I didn't see them in the obvious test/jdk/java/lang/Class directory.

They are located in `test/jdk/java/lang/constant/TypeDescriptorTest.java`.

Reply:
> > Did you consider calling getComponentType(), which already returns the field?
> 
> I don't think we want an extra indirection here - The logic here is quite simple. Counterargument could be that `MethodType::descriptorString` calls `toMethodDescriptorString`, except these methods involve more complex caching and is not simple like this field access.

My mental model anyway is that this trivial inline would be something the VM would do readily.